task/buildah-rhtap/0.1/buildah-rhtap.yaml

```diff
@@ -0,0 +1,237 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.1"
+    build.appstudio.redhat.com/build_type: "docker"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "image-build, appstudio, hacbs"
+  name: buildah-rhtap
```

Comment:
Sure, if you want the long version, here's my [attempt](https://github.com/redhat-appstudio/build-definitions/commit/1e548f9709015d4a0cd0ea2ffe83b9feadfddbb0) to port the task into build-definitions as it is, and a [version](https://github.com/redhat-appstudio/build-definitions/commit/4466dc6f3fb93fa1d18407ca8a3500c7a6b609af) with the SBOM generation added to it. 

But in summary, the default buidah task:
- validates input params, existence of the Dockerfile and build context
- builds the image
    - can use a dockerconfig secret (optional) 
    - sets the --no-cache option
    - allows for extra args build args to be added
- pushes image 
    - allows for extra push args to be added
    - can be skipped
- emits the results
    - digest sum
    - image name and tag

It is much more straightforward, it doesn't use things such as root access or a mounted `/var/lib/containers` volume, as it does build an push in the same step.

Reply:
> can use a dockerconfig secret (optional)

I think ours does too, but implicitly (tekton automatically injects some secrets to predefined directories)

> allows for extra args build args to be added

As in `--build-arg`? Or any argument at all? That would give the default buildah task way more flexibility. Proably too much flexibility to be EC-verifiable, since the user could make buildah do just about anything with the extra args

Reply:
Any argument at all... this is what it looks like:
```
buildah \
      --storage-driver="${PARAMS_STORAGE_DRIVER}" \
      --tls-verify="${PARAMS_TLS_VERIFY}" \ 
      bud ${PARAMS_BUILD_EXTRA_ARGS} \
            --no-cache \
            --file="${DOCKERFILE_FULL}" \
            --tag="${PARAMS_IMAGE}" \
            ${PARAMS_CONTEXT}
 ```

Reply:
Yeah, this would allow them to mount any volumes and secrets and who knows what else. Which we may need to support in the future, but probably in a more controlled way